### PR TITLE
#3986 [Manager] Implement autofill

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3986 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/utils/validate.js [hunk 1]
```diff
@@ -61,8 +61,11 @@ export const checkDuplicate = validateList => value => {
   );
 };
 
-export const composeValidators = (...validators) => value =>
-  validators.reduce((error, validator) => error || validator(value), undefined);
+export const composeValidators = (...validators) => (value, allValues, meta) =>
+  validators.reduce(
```
Comment (konekoya) [inline ohara-manager/client/src/utils/validate.js]: 👍

**Comment by saivirtue** [comment]
The `AutofillList` could not get the latest settings when **first created workspace**, could you take a look of that?

**Comment by eechih** [comment]
> The AutofillList could not get the latest settings when first created workspace

@saivirtue Thanks for reporting. I will take a look :)

File: ohara-manager/client/src/context/api/workspaceApi.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ export const createApi = context => {
       }
       return map(res.data, object => {
         return generateClusterResponse({
-          values: object.tags,
+          values: { ...object.tags, ...getKey(object) },
```
Comment (eechih) [inline ohara-manager/client/src/context/api/workspaceApi.js]: @saivirtue Api 回傳資料缺少 group, name，這是新增 Autofill 而沒有出現在 List 的主要原因。

**Pull request merged**